.travis.yml

```diff
@@ -1,9 +1,35 @@
+sudo: required
+
 language: java
+
+services:
+  - docker
+
+jdk:
+  - oraclejdk8
+
 before_cache:
-- rm -f  $HOME/.gradle/caches/modules-2/modules-2.lock
-- rm -fr $HOME/.gradle/caches/*/plugin-resolution/
+  - rm -f  $HOME/.gradle/caches/modules-2/modules-2.lock
+  - rm -fr $HOME/.gradle/caches/*/plugin-resolution/
+
 cache:
   directories:
-  - $HOME/.gradle/caches/
-  - $HOME/.gradle/wrapper/
+    - $HOME/.gradle/caches/
+    - $HOME/.gradle/wrapper/
+
+branches:
+  only:
+    - master
+    - /^feature\/[a-zA-Z0-9\-]+$/
```

Comment:
do we need this? What about 'bug' and 'refactor' branches for example?

Reply:
saw few branches out there with `feature` prefix. if there is a general agreement per team how to do code reviews, travis builds etc, there can be more reasonable branch limitations there.
this is merely travis build limitations per branch(-es) as oppose to what it was before - build every single time commit has been pushed

Reply:
Can we set it up on travis instead? Ie, build on pull requests only.

Reply:
it is set up. on the other hand, one should not trust in third party tool configuration and have it present in each codebase you wish to, applying custom config as needed.
for example, i don't want to build CI for any subsequent branches because my implementation requires so but only against enlisted ones

Reply:
Seems like fork-based workflow is coming.